code/modules/mob/living/carbon/human/human_defense.dm

```diff
@@ -499,48 +501,51 @@
 		arm_clothes = w_uniform
 	if(wear_suit && (wear_suit.body_parts_covered & HANDS) || wear_suit && (wear_suit.body_parts_covered & ARMS))
 		arm_clothes = wear_suit
-	if(arm_clothes)
-		if(!arm_clothes.unacidable)
-			arm_clothes.acid_act(acidpwr, acid_volume_left)
-			acid_volume_left = max(acid_volume_left - acid_decay, 0)
-			update_inv_gloves()
-			update_inv_w_uniform()
-			update_inv_wear_suit()
+
+	if(!bodyzone_hit || bodyzone_hit == "l_arm" || bodyzone_hit == "r_arm")
+		if(arm_clothes)
+			if(arm_clothes.acid_state != UNACIDABLE)
+				arm_clothes.acid_act(acidpwr, acid_volume_left)
+				acid_volume_left = max(acid_volume_left - acid_decay, 0)
+				update_inv_gloves()
+				update_inv_w_uniform()
+				update_inv_wear_suit()
+			else
+				src << "<span class='notice'>Your [arm_clothes.name] protects your arms and hands from the acid!</span>"
 		else
-			src << "<span class='notice'>Your [arm_clothes.name] protects your arms and hands from the acid!</span>"
-	else
-		. = get_bodypart("r_arm")
-		if(.)
-			damaged += .
-		. = get_bodypart("l_arm")
-		if(.)
-			damaged += .
+			. = get_bodypart("r_arm")
+			if(.)
+				damaged += .
+			. = get_bodypart("l_arm")
+			if(.)
+				damaged += .
 
 
 	//LEGS & FEET//
 	var/obj/item/clothing/leg_clothes = null
 	if(shoes)
 		leg_clothes = shoes
-	if(w_uniform && (w_uniform.body_parts_covered & FEET) || w_uniform && (w_uniform.body_parts_covered & LEGS))
+	if(w_uniform && ((w_uniform.body_parts_covered & FEET) || (bodyzone_hit != "feet" && (w_uniform.body_parts_covered & LEGS))))
 		leg_clothes = w_uniform
-	if(wear_suit && (wear_suit.body_parts_covered & FEET) || wear_suit && (wear_suit.body_parts_covered & LEGS))
+	if(wear_suit && ((wear_suit.body_parts_covered & FEET) || (bodyzone_hit != "feet" && (wear_suit.body_parts_covered & LEGS))))
 		leg_clothes = wear_suit
-	if(leg_clothes)
-		if(!leg_clothes.unacidable)
-			leg_clothes.acid_act(acidpwr, acid_volume_left)
-			acid_volume_left = max(acid_volume_left - acid_decay, 0)
-			update_inv_shoes()
-			update_inv_w_uniform()
-			update_inv_wear_suit()
+	if(!bodyzone_hit || bodyzone_hit == "l_leg" || bodyzone_hit == "r_leg")

```

Comment:
you used ```"feet"``` earlier for walking on acid, but it doesn't appear to be covered anywhere? (it <i>alters</i> how leg_clothes are handled, but is never itself handled)